pkg/asset/tls/ironictls.go

```diff
@@ -0,0 +1,55 @@
+package tls
+
+import (
+	"context"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"net"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+)
+
+// IronicTLSCert is the asset that generates the key/cert pair that is used for
+// enabling TLS for virtual media in ironic.
+type IronicTLSCert struct {
+	SelfSignedCertKey
+}
+
+var _ asset.WritableAsset = (*IronicTLSCert)(nil)
+
+// Dependencies returns the dependency of the the cert/key pair, which includes
+// the parent CA, and install config if it depends on the install config for
+// DNS names, etc.
+func (a *IronicTLSCert) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.InstallConfig{},
+	}
+}
+
+// Generate generates the cert/key pair based on its dependencies.
+func (a *IronicTLSCert) Generate(ctx context.Context, dependencies asset.Parents) error {
+	installconfig := &installconfig.InstallConfig{}
+	dependencies.Get(installconfig)
+	apiVIP := installconfig.AssetBase.Config.BareMetal.APIVIPs[0]
+	logrus.Debugf("API VIP: %s", apiVIP)
+	var ipAddresses []net.IP
+	ipAddresses = append(ipAddresses, net.ParseIP(apiVIP))
+
+	cfg := &CertCfg{
+		Subject:      pkix.Name{CommonName: apiVIP},
```

Comment:
We've discovered that using an IPv6 address here fails validation. Since CommonName is no longer used for anything meaningful, let's revert to "ironic" or something like that.

Reply:
Done